airflow/api_fastapi/common/parameters.py

```diff
@@ -281,6 +282,30 @@ def _safe_parse_datetime(date_to_check: str) -> datetime:
         )
 
 
+class _WarningTypeFilter(BaseParam[str]):
+    """Filter on warning type."""
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None and self.skip_none:
+            return select
+        return select.where(DagWarning.warning_type.contains(self.value))
```

Comment:
Why a 'contains' here ? `warning_type` is a `str` not an array.

And in the legacy code it is an equal check `query.where(DagWarningModel.warning_type == warning_type)`

Reply:
fixed it